src/ext-test/java/org/opentripplanner/ext/legacygraphqlapi/GraphQLApiTest.java

```diff
@@ -64,7 +96,31 @@ class GraphQLApiTest {
               .build()
           )
       );
-    context = TestServerContext.createServerContext(graph, transitModel);
+
+    Itinerary i1 = newItinerary(A, T11_00)
+      .walk(20, B)
+      .bus(122, T11_01, T11_15, C)
+      .rail(439, T11_30, T11_50, D)
+      .build();
+
+    var railLeg = i1.getTransitLeg(2);
+    var alert = TransitAlert
+      .of(id("an-alert"))
+      .withHeaderText(new NonLocalizedString("A header"))
+      .withDescriptionText(new NonLocalizedString("A description"))
+      .withUrl(new NonLocalizedString("https://example.com"))
+      .withCause(AlertCause.MAINTENANCE)
+      .withEffect(AlertEffect.REDUCED_SERVICE)
+      .withSeverity(AlertSeverity.VERY_SEVERE)
+      .addEntity(new EntitySelector.Stop(stop.getId()))
+      .addTimePeriod(
+        new TimePeriod(ALERT_START_TIME.getEpochSecond(), ALERT_END_TIME.getEpochSecond())
+      )
+      .build();
+    railLeg.addAlert(alert);
+
+    context = new TestServerRequestContext(graph, transitModel);
+    context.setRoutingResult(List.of(i1));
```

Comment:
I'm a bit reluctant about this, would there be a way to use (and expand) the `TestServerContext`?

Reply:
Also now we will get errors if we try to query information about the stops on the itinerary, as the stops `B`, `C` and `D` are not part of the stop model.

Reply:
What I'm trying to avoid is this becoming a full-blown integration test that does actual routing. That's why I went with this approach.

Do you have a suggestion on how to achieve that?

Or do you think it _should_ be an integration test?

Reply:
It is already an integration test. If we want to do unit testing, it should be done for the `LegacyGraphQL____Impl` classes, by passing `DataFetchingEnvironment`s to the various `DataFetcher`s

Probably the easiest is to just manually create a minimal coherent transit network, where all the fields resolve to non-nullable types where we have such in the "real" data model. Injecting can be done, by inversion of control, and being able to pass in a `LegacyGraphQLRequestContext` from the outside. That way we can mock just the routing service.

Reply:
Also, we should move from the legacy `context` to using a `GraphQLContext`, but that is a separate task

Reply:
Ok, let me rephrase that. I would prefer not to test the routing engine as there are lots of other tests for that.

Perhaps it would be a good thing to pair-program on that. Would you be interested?

My main motivation is simply to get the code coverage up and to have a test-based development workflow. If we achieve that I'm happy to implement this.

Reply:
Sure, I have time tomorrow before the OTP dev meeting.

In order to have a test, we need to decide what we want to test. If the scope of the test is a set of resolvers, then we should have tests on the individual `DataFetcher` implementations. Now we also test the JAX-RS interface and `LegacyGraphQLIndex` setup, which should be out of scope, if we want to do proper unit testing.

There is no problem in mocking the `RoutingService` in order to return a static result. However in that case we won't do testing on the mapping on the input parameters, which is ok, since the mapping should be its is own unit. This can be tested, by feeding a custom `DataFetchingEnvironment` containing various combinations of parameters to a method extracted from `LegacyGraphQLQueryTypeImpl#plan` and asserting the created `RouteRequest` contains the correct fields.